src/FsCodec.SystemTextJson/UnionConverter.fs

```diff
@@ -25,3 +29,111 @@ module private Union =
                 caseConstructor = cases |> Array.map (fun c -> FSharpValue.PreComputeUnionConstructor(c, true))
             } |> Some
     let tryGetUnion : Type -> Union option = memoize _tryGetUnion
+
+
+    /// Paralells F# behavior wrt how it generates a DU's underlying .NET Type
+    let inline isInlinedIntoUnionItem (t : Type) =
+        t = typeof<string>
+        || t.IsValueType
+        || t.IsArray
+        || (t.IsGenericType
+           && (typedefof<Option<_>> = t.GetGenericTypeDefinition()
+                || t.GetGenericTypeDefinition().IsValueType)) // Nullable<T>
+
+    let typeHasJsonConverterAttribute = memoize (fun (t : Type) -> t.IsDefined(typeof<Serialization.JsonConverterAttribute>, false))
+
+    let propTypeRequiresConstruction (propertyType : Type) =
+        not (isInlinedIntoUnionItem propertyType)
+        && not (typeHasJsonConverterAttribute propertyType)
+
+    /// Prepare arguments for the Case class ctor based on the kind of case and how F# maps that to a Type
+    /// and/or whether we need to let json.net step in to convert argument types
+    let mapTargetCaseArgs (element : JsonElement) options : PropertyInfo[] -> obj [] = function
+        | [| singleCaseArg |] when propTypeRequiresConstruction singleCaseArg.PropertyType ->
+            [| JsonSerializer.DeserializeElement (element, singleCaseArg.PropertyType, options) |]
+        | multipleFieldsInCustomCaseType ->
+            [| for fi in multipleFieldsInCustomCaseType ->
+                match element.GetProperty fi.Name with
+                // TOTHINK: I'm not sure this applies with STJ
+
+                //| null when
+                //    // Afford converters an opportunity to handle the missing field in the best way I can figure out to signal that
+                //    // The specific need being covered (see tests) is to ensure that, even with MissingMemberHandling=Ignore,
+                //    // the TypeSafeEnumConverter should reject missing values
+                //    // not having this case would go direct to `null` without passing go
+                //    typeHasJsonConverterAttribute fi.PropertyType
+                //    || serializer.MissingMemberHandling = MissingMemberHandling.Error ->
+                //        // NB caller can opt out of erroring by setting NullValueHandling = NullValueHandling.Ignore)
+                //        // which renders the following equivalent to the next case
+                //        JToken.Parse("null").ToObject(fi.PropertyType, serializer)
```

Comment:
TODO: Remove MissingMemberHandling case (and associated tests?)

Reply:
Yeah, supporting that feature is quite an edge case
unless there's some complete and straightforward porting guide, it can definitely be parked

Ways of parking:
- log an issue with an up-for-grabs (more valuable if you can picture someone doing that and have no other such items)
- put a one-liner note at top of file noting discrepancy with regard to Newtonsoft version

probably leave a TOCONSIDER comment to replace the tests

Reply:
Yeah, I don't think we'll be able to get this one back:

>#### MissingMemberHandling
>Newtonsoft.Json can be configured to throw exceptions during deserialization if the JSON includes properties that are missing in the target type. System.Text.Json ignores extra properties in the JSON, except when you use the [JsonExtensionData] attribute. **There's no workaround for the missing member feature** [emphasis mine].